### PR TITLE
Gradeable model refactor electronic 0 ajax 2 conflict saving

### PR DESCRIPTION
**Pull request by jchulton**
Saving now works with the new model

### CODE REVIEW

File: site/app/libraries/Access.php [hunk 1]
```diff
@@ -85,7 +85,7 @@ public function __construct(Core $core) {
         $this->permissions["grading.electronic.grade.if_no_sections_exist"] = self::ALLOW_MIN_INSTRUCTOR;
         $this->permissions["grading.electronic.save_graded_component"] = self::ALLOW_MIN_STUDENT | self::CHECK_GRADEABLE_MIN_GROUP | self::CHECK_GRADING_SECTION_GRADER | self::CHECK_PEER_ASSIGNMENT_STUDENT | self::CHECK_HAS_SUBMISSION | self::CHECK_COMPONENT_PEER_STUDENT;
         $this->permissions["grading.electronic.save_general_comment"] = self::ALLOW_MIN_STUDENT | self::CHECK_GRADEABLE_MIN_GROUP | self::CHECK_GRADING_SECTION_GRADER | self::CHECK_PEER_ASSIGNMENT_STUDENT | self::CHECK_HAS_SUBMISSION;
-        $this->permissions["grading.electronic.get_mark_data"] = self::ALLOW_MIN_STUDENT | self::CHECK_GRADEABLE_MIN_GROUP | self::CHECK_GRADING_SECTION_GRADER | self::CHECK_PEER_ASSIGNMENT_STUDENT | self::CHECK_COMPONENT_PEER_STUDENT;
+        $this->permissions["grading.electronic.get_mark_data"] = self::ALLOW_MIN_LIMITED_ACCESS_GRADER;
```
Comment (KevinMackenzie) [inline site/app/libraries/Access.php]: remove this modification

File: site/app/libraries/database/DatabaseQueries.php [hunk 2]
```diff
@@ -1339,6 +1339,7 @@ public function insertGradeableComponentData($gd_id, GradeableComponent $compone
     /**
      * @param string             $gd_id
      * @param string             $grader_id
+     * @param string             $verifier_id
```
Comment (KevinMackenzie) [inline site/app/libraries/database/DatabaseQueries.php]: also this

File: site/public/js/ta-grading-mark.js [hunk 3]
```diff
@@ -328,17 +328,19 @@ function ajaxSaveGradedComponent(gradeable_id, component_id, anon_id, active_ver
     })
 }
 
-function ajaxSaveMark(gradeable_id, component_id, mark_id, points, note, async, successCallback, errorCallback) {
+function ajaxSaveMark(gradeable_id, user_id, component_id, mark_id, points, note, async, successCallback, errorCallback) {
```
Comment (KevinMackenzie) [inline site/public/js/ta-grading-mark.js]: remove modifications to this function

File: site/app/controllers/admin/AdminGradeableController.php [hunk 4]
```diff
@@ -243,7 +243,7 @@ private function editPage(Gradeable $gradeable, $semester, $course, $nav_tab = 0
 
     /* Http request methods (i.e. ajax) */
 
-    private function tryGetGradeable($gradeable_id) {
+    protected function tryGetGradeable(string $gradeable_id, bool $render_json = true) {
```
Comment (KevinMackenzie) [inline site/app/controllers/admin/AdminGradeableController.php]: What is this change doing?

File: migration/migrations/course/20180723124511_regrade_per_assignment.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+def up(config, conn, semester, course):
```
Comment (KevinMackenzie) [inline migration/migrations/course/20180723124511_regrade_per_assignment.py]: migration file here

File: site/app/libraries/Access.php [hunk 6]
```diff
@@ -85,8 +85,7 @@ public function __construct(Core $core) {
...
         $this->permissions["grading.electronic.save_general_comment"] = self::ALLOW_MIN_STUDENT | self::CHECK_GRADEABLE_MIN_GROUP | self::CHECK_GRADING_SECTION_GRADER | self::CHECK_PEER_ASSIGNMENT_STUDENT | self::CHECK_HAS_SUBMISSION;
-        $this->permissions["grading.electronic.get_mark_data"] = self::ALLOW_MIN_STUDENT | self::CHECK_GRADEABLE_MIN_GROUP | self::CHECK_GRADING_SECTION_GRADER | self::CHECK_PEER_ASSIGNMENT_STUDENT | self::CHECK_COMPONENT_PEER_STUDENT;
-        $this->permissions["grading.electronic.get_gradeable_comment"] = self::ALLOW_MIN_STUDENT | self::CHECK_GRADEABLE_MIN_GROUP | self::CHECK_GRADING_SECTION_GRADER | self::CHECK_PEER_ASSIGNMENT_STUDENT;
+        $this->permissions["grading.electronic.get_mark_data"] = self::ALLOW_MIN_STUDENT | self::CHECK_GRADEABLE_MIN_GROUP | self::CHECK_GRADING_SECTION_GRADER | self::CHECK_PEER_ASSIGNMENT_STUDENT | self::CHECK_COMPONENT_PEER_STUDENT;        $this->permissions["grading.electronic.get_gradeable_comment"] = self::ALLOW_MIN_STUDENT | self::CHECK_GRADEABLE_MIN_GROUP | self::CHECK_GRADING_SECTION_GRADER | self::CHECK_PEER_ASSIGNMENT_STUDENT;
```
Comment (KevinMackenzie) [inline site/app/libraries/Access.php]: I think you just deleted a newline by accident here

File: site/public/js/ta-grading-mark.js [hunk 7]
```diff
@@ -336,9 +337,10 @@ function ajaxSaveMark(gradeable_id, component_id, mark_id, points, note, async,
         data: {
             'gradeable_id' : gradeable_id,
             'component_id' : component_id,
+            'gradeable_component_id' : component_id,
```
Comment (KevinMackenzie) [inline site/public/js/ta-grading-mark.js]: There are a few other changes to this function that still aren't reverted

File: site/public/js/ta-grading-mark.js [hunk 8]
```diff
@@ -350,40 +352,10 @@ function ajaxSaveMark(gradeable_id, component_id, mark_id, points, note, async,
                 successCallback(response);
             }
         },
-        error: errorCallback
+        error: successCallback
```
Comment (KevinMackenzie) [inline site/public/js/ta-grading-mark.js]: This doesn't seem very intuitive to me...

File: site/app/controllers/admin/AdminGradeableController.php (hunk 4 above)
Comment (jchulton) [reply site/app/controllers/admin/AdminGradeableController.php]: In AbstractController, trygetGradeable has the protected method header. Giving the function the wrong method header results in an on page exception.

**Pull request merged**